dev/src/reference.ts

```diff
@@ -1821,6 +1804,25 @@ export class Query<T = firestore.DocumentData> implements firestore.Query<T> {
     );
   }
 
+  /**
+   * Returns the sorted array of inequality filter fields used in this query.
```

Comment:
We should use consistent implementation when porting between platforms, unless there's a compelling reason not to. IIUC other platforms return a Set with unique inequality filter fields, but this implementation returns an array with potentially duplicate fields. Is there a reason?

We should either use set in all platforms, or an array with potential duplicates on all platforms.

Reply:
In web SDK, we have custom class `SortedSet` which allows us to use custom comparator, ie, the `FieldPath.compareTo()`. But in node sdk, we do not have that `SortedSet`. And `Set` in javascript replies on its own build in equality comparison, which is based on object identity. 

In another word, is we use Set<FieldPath>, `getInequalityFilterFields` on this query will end up with a set of size 2.
`collection
        .where('a', '<', 5)
        .where('a', '>', 1)
`

Should I introduce SortedSet to node SDK? 🤔  It is useful in some cases and not that big: https://github.com/firebase/firebase-js-sdk/blob/78d2738c246555556cba8dcfe2932639f80523ea/packages/firestore/src/util/sorted_set.ts#L27

Reply:
I didn't realize this SDK doesn't have `SortedSet`. I think it's probably overkill to copy that class into this SDK just for this purpose given that this is the only usage and the caller here has to use a Set to avoid duplicates anyway.

No action needed.